### PR TITLE
Add HuggingFace ViT tests

### PR DESCRIPTION
**Pull request by RissyRan**
* Integrate with HuggingFace to add ViT tests for JAX
* Add functional tests for v2, v3, v4 and convergence test for v4-32
* Tests:
  * flax-latest-hf-vit-func-v2-8-1vm-rht8t
  * flax-latest-hf-vit-func-v2-32-1vm-flxg
  * flax-latest-hf-vit-conv-v4-32-1vm-72z9k

### CODE REVIEW

File: tests/jax/latest/hf-vit.libsonnet [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+  local func_tests_v2_v3 = [
+    hf_vit_common + hf_vit_func_v2_v3 + accelerator
+    for accelerator in [v2_8, v2_32, v3_8, v3_32]
+  ],
+  local func_tests_v4 = [
+    hf_vit_common + hf_vit_func_v4 + accelerator
+    for accelerator in [v4_8, v4_32]
+  ],
+  local conv_tests_v4 = [hf_vit_common + hf_vit_conv_v4 + v4_32],
```
Comment (allenwang28) [inline tests/jax/latest/hf-vit.libsonnet:125]: Building on top of a previous comment, I think it makes more sense to have

func_tests
conv_tests

which includes v2, v3, v4.

**Review by allenwang28 — COMMENTED** [review]
Nice work! Some comments

File: tests/jax/latest/hf-vit.libsonnet [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+      %(testEnvWorkarounds)s
+      %(installLatestJax)s
+
+      python3 -c 'import flax; print("flax version:", flax.__version__)'
+      num_devices=`python3 -c "import jax; print(jax.device_count())"`
+      if [ "$num_devices" = "1" ]; then
+        echo "No TPU devices detected"
+        exit 1
+      fi
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply tests/jax/latest/hf-vit.libsonnet:48]: Sure thing, will do that!

File: tests/jax/latest/hf-vit.libsonnet (hunk 1 above)
Comment (RissyRan) [reply tests/jax/latest/hf-vit.libsonnet:125]: Sounds good, will update that.

File: tests/jax/latest/hf-vit.libsonnet [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                },
+                inclusive_bounds: false,
+                wait_for_n_data_points: 0,
+              },
+            },
+          },
+        },
+      },
+    },
+  },
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply tests/jax/latest/hf-vit.libsonnet:96]: OK, sounds good!

File: tests/jax/latest/hf-vit.libsonnet [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+      wget https://s3.amazonaws.com/fast-ai-imageclas/imagenette2.tgz
+      tar -xvzf imagenette2.tgz
+
+      export GCS_BUCKET=$(MODEL_DIR)
+      python3 examples/flax/vision/run_image_classification.py \
+        --output_dir './vit-imagenette' \
+        --train_dir='imagenette2/train' \
+        --validation_dir='imagenette2/val' \
+        --learning_rate 1e-3 \
+        --preprocessing_num_workers 32 \
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply tests/jax/latest/hf-vit.libsonnet:59]: It could be other numbers I think. 32 is a recommended number from example in hugging face.

**Pull request merged**